include/maths/CDataFrameUtils.h

```diff
@@ -59,52 +62,115 @@ class MATHS_EXPORT CDataFrameUtils : private core::CNonInstantiatable {
     using TSizeDoublePr = std::pair<std::size_t, double>;
     using TSizeDoublePrVec = std::vector<TSizeDoublePr>;
     using TSizeDoublePrVecVec = std::vector<TSizeDoublePrVec>;
+    using TSizeDoublePrVecVecVec = std::vector<TSizeDoublePrVecVec>;
     using TRowRef = core::CDataFrame::TRowRef;
     using TWeightFunction = std::function<double(TRowRef)>;
     using TQuantileSketchVec = std::vector<CQuantileSketch>;
     using TPackedBitVectorVec = std::vector<core::CPackedBitVector>;
 
     //! \brief Used to extract the value from a specific column of the data frame.
-    class CColumnValue {
+    class MATHS_EXPORT CColumnValue {
     public:
+        CColumnValue(std::size_t column) : m_Column{column} {}
         virtual ~CColumnValue() = default;
         virtual double operator()(const TRowRef& row) const = 0;
         virtual double operator()(const TFloatVec& row) const = 0;
+        virtual std::size_t id() const = 0;
```

Comment:
Changed to hash (since it is used to key a hash set) in [this](https://github.com/elastic/ml-cpp/pull/591/commits/3982962d91b31d8d0f2950683820a3563cce1b49) commit.